certora/specs/SafeMath.spec

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+definition WAD() returns uint256 = 10^18;
+
+function summaryWMulDown(uint256 x,uint256 y) returns uint256 {
+    // Safe require because the reference implementation would revert.
+    return require_uint256((x * y)/WAD());
+}
+
+function summaryWDivUp(uint256 x,uint256 y) returns uint256 {
+    // Safe require because the reference implementation would revert.
+    return require_uint256((x * WAD() + (y-1)) / y);
+}
+
+
+// Check that LTV <= LLTV is equivalent to borrowed <= (collateralQuoted * LLTV) / WAD.
+rule borrowedLECollatQuotedTimesLLTVEqLtvLTEqLLTV {
+    uint256 borrowed;
+    uint256 collateralQuoted;
+
+    // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
+    require collateralQuoted > 0;
+
+    mathint ltv = summaryWDivUp(borrowed, collateralQuoted);
+
+    assert (ltv <= currentContract.LLTV) <=> borrowed <= summaryWMulDown(collateralQuoted, currentContract.LLTV);
+}
+
+// Check that substracting the PRE_LLTV to LTV wont underflow.
+rule ltvMinusPreLltvWontUnderflow {
+    uint256 borrowed;
+    uint256 collateralQuoted;
+    uint256 preLltv;
+
+    // Safe require because the implementation would revert, see rule zeroCollateralQuotedReverts.
+    require (collateralQuoted > 0);
+
+    // Safe require because the implementation would revert if borrowed threshold is not ensured.
+    uint256 borrowThreshold = summaryWMulDown(collateralQuoted, preLltv);
+    require (borrowed > borrowThreshold);
+
+    uint256 ltv = summaryWDivUp(borrowed, collateralQuoted);
+    assert ltv >= preLltv;
```

Comment:
I'm confused of where do you want to check which equivalence, do you mean here in `SafeMath`? Or in `Reverts.spec`?

Reply:
Yes, here. The other rule used to be an equivalence, so it made sense to have this rule be an equivalence as well

Reply:
But here `ltv` ins't equal to `preLltv`, is it?

Reply:
No it's not, I'm not sure why you are asking this. But here what I'm suggesting: instead of doing `require A` and then `assert B`, we do `assert A <=> B`

Reply:
It wasn't clear, this formulation removes ambiguities thank you 😃

Reply:
Done in https://github.com/morpho-org/pre-liquidation/pull/89/commits/c715feb778826566a40deb4543be216a1c98523e.